skt/reporter.py

```diff
@@ -470,6 +470,12 @@ def __getjobresults(self):
         minfo = {"short": {}, "long": {}}
         jidx = 1
         for jobid, values in vresults.items():
+            job_result = values.get('result')
+            job_status = values.get('status')
+            if job_result == 'Warn' and job_status == 'Aborted':
+                logging.info('Skipping aborted job %s', jobid)
+                continue
+
```

Comment:
Can we follow similar logic here? Accept only particular results as valid, and drop the rest?

Also, what happens to the dropped job? I would say that a single (errored) aborted job invalidates the whole run. I.e. we basically skipped one test and we don't know its outcome. We should re-run that job. Do we do that, or do we send a report after that?

Reply:
The resubmission of a single aborted job is already done (by that code that tries to rule out host-specific failures). What we are doing here is just not reporting it since it's not useful for the developer. Does that answer you question?

Reply:
Rriiight, so do you mean that we still send the message, but don't include this job's result into the report?

Reply:
Or rather, does skt return 1 or 2, if an aborted (failed due to infrastructure) job is detected?

Reply:
> Rriiight, so do you mean that we still send the message, but don't include this job's result into the report?

Yes.

> Or rather, does skt return 1 or 2, if an aborted (failed due to infrastructure) job is detected?

Do you mean the runner stage now? If all the jobs abort, 2. If the aborts are host-specific, then 0 (as per the host-specific resubmission logic)

Reply:
How do we detect host-specific failure?

Reply:
The resubmission logic is pretty confusing and I didn't dive into it, but here's what I gathered so far:

For all non-pass jobs, the host and failure data are saved. By default, all these jobs are then resubmitted to a random machine from the pool, as well as the same host which the job failed on. When the testing results are gathered (after all the jobs finished), the failure data is checked and if there's only one unique hostname for the job, it's a host-specific problem.

Reply:
I'm thinking that an infrastructure failure shouldn't even count into the host-specific failure detection. I.e. it should be as if the job simply didn't run, and it should be restarted, no matter if it's the first job in the host failure detection routine, or any other one.

But to be sure, what exactly are we calling infrastructure failures here? What would be the actual reason for the condition we're checking here for?

Reply:
> But to be sure, what exactly are we calling infrastructure failures here? What would be the actual reason for the condition we're checking here for?

I'd say things like "network is down so we can't connect anywhere", "Beaker is under maintenance so we can't submit any jobs", "somebody messed with lab controller and now all jobs abort before even starting" etc.

The fact that a single job aborts is IMHO not an infrastructure failure. It's annoying, sure, but that's why we have the check there.

Reply:
Ah, I see, good, those all seem to be perfect "infrastructure failures", indeed.

What I'm saying is that those shouldn't be the basis for host failure check. Because the fact that such failure repeats or doesn't repeat is not saying anything about the host. I would say we should then simply resubmit the job exactly as it is (whether it's a part of a host failure check or not), and if e.g. we get three infrastructure failures in a row, then give up with exit status of 2.

Reply:
Oh, I see what you mean, I think. I'll add a check for that as well.